ibm/service/codeengine/resource_ibm_code_engine_secret.go

```diff
@@ -0,0 +1,328 @@
+
+	parts, err := flex.SepIdParts(d.Id(), "/")
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	getSecretOptions.SetProjectID(parts[0])
+	getSecretOptions.SetName(parts[1])
+
+	secret, response, err := codeEngineClient.GetSecretWithContext(context, getSecretOptions)
+	if err != nil {
+		if response != nil && response.StatusCode == 404 {
+			d.SetId("")
+			return nil
+		}
+		log.Printf("[DEBUG] GetSecretWithContext failed %s\n%s", err, response)
+		return diag.FromErr(fmt.Errorf("GetSecretWithContext failed %s\n%s", err, response))
+	}
+
+	if err = d.Set("project_id", secret.ProjectID); err != nil {
+		return diag.FromErr(fmt.Errorf("Error setting project_id: %s", err))
+	}
+	if err = d.Set("format", secret.Format); err != nil {
+		return diag.FromErr(fmt.Errorf("Error setting format: %s", err))
+	}
+	if err = d.Set("name", secret.Name); err != nil {
+		return diag.FromErr(fmt.Errorf("Error setting name: %s", err))
+	}
+	if !core.IsNil(secret.Data) {
+		data := make(map[string]string)
+		for k, v := range secret.Data {
+			data[k] = string(v)
+		}
+		if err = d.Set("data", data); err != nil {
+			return diag.FromErr(fmt.Errorf("Error setting data: %s", err))
+		}
+	}
+	if !core.IsNil(secret.CreatedAt) {
+		if err = d.Set("created_at", secret.CreatedAt); err != nil {
+			return diag.FromErr(fmt.Errorf("Error setting created_at: %s", err))
+		}
+	}
+	if err = d.Set("entity_tag", secret.EntityTag); err != nil {
+		return diag.FromErr(fmt.Errorf("Error setting entity_tag: %s", err))
+	}
+	if !core.IsNil(secret.Href) {
+		if err = d.Set("href", secret.Href); err != nil {
+			return diag.FromErr(fmt.Errorf("Error setting href: %s", err))
+		}
+	}
+	if !core.IsNil(secret.ID) {
+		if err = d.Set("id", secret.ID); err != nil {
+			return diag.FromErr(fmt.Errorf("Error setting id: %s", err))
+		}
+	}
+	if !core.IsNil(secret.ResourceType) {
+		if err = d.Set("resource_type", secret.ResourceType); err != nil {
+			return diag.FromErr(fmt.Errorf("Error setting resource_type: %s", err))
+		}
+	}
+	if !core.IsNil(secret.ID) {
+		if err = d.Set("secret_id", secret.ID); err != nil {
+			return diag.FromErr(fmt.Errorf("Error setting secret_id: %s", err))
+		}
+	}
+	if err = d.Set("etag", response.Headers.Get("Etag")); err != nil {
+		return diag.FromErr(fmt.Errorf("Error setting etag: %s", err))
+	}
+
+	return nil
+}
+
+func resourceIbmCodeEngineSecretUpdate(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	codeEngineClient, err := meta.(conns.ClientSession).CodeEngineV2()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	replaceSecretOptions := &codeenginev2.ReplaceSecretOptions{}
+
+	parts, err := flex.SepIdParts(d.Id(), "/")
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	replaceSecretOptions.SetProjectID(parts[0])
+	replaceSecretOptions.SetName(parts[1])
+	replaceSecretOptions.SetFormat(d.Get("format").(string))
+
+	hasChange := false
+
+	if d.HasChange("project_id") {
```

Comment:
Same as above  ForceNew controls this behaviour

Reply:
Removed